### PR TITLE
Refactoring Credentials and adding Scope

### PR DESCRIPTION
**Pull request by stjohnjohnson**
Also added pretty-printing of errors (no more 500 with no context).

### CODE REVIEW

File: lib/server.js [hunk 1]
```diff
@@ -4,6 +4,29 @@ const Hapi = require('hapi');
...
+ * If we're throwing errors, let's have them say a little more than just 500
+ * @method prettyPrintErrors
+ * @param  {Hapi.Request}    request Hapi Request object
+ * @param  {Hapi.Reply}      reply   Hapi Reply object
+ */
+function prettyPrintErrors(request, reply) {
+    if (request.response.isBoom) {
+        const err = request.response;
+        const errName = err.output.payload.error;
+        const errMessage = err.message;
```
Comment (petey) [inline lib/server.js]: What is the difference between `err.message` and `err.output.payload.message`? Can this method just `return reply(err.output.payload).code(statusCode)`?

Comment (stjohnjohnson) [inline lib/server.js]: Boom will automatically hide all 500 errors from the end user.  https://github.com/hapijs/boom/blob/master/lib/index.js#L136-L138

This reverts that feature.

Comment (petey) [inline lib/server.js]: Got it. 😦  Lame.

**Comment by petey** [comment]
👍 

cat_gif

**Comment by jer** [comment]
👍

File: lib/server.js (hunk 1 above)
Comment (FenrirUnbound) [inline lib/server.js]: It's also [in their docs](https://github.com/hapijs/boom#http-5xx-errors)

> All 500 errors hide your message from the end user. Your message is recorded in the server log.

**Comment by FenrirUnbound** [comment]
👍 


ckhe4w1 - imgur

**Pull request merged**